typescript/utils.ts

```diff
@@ -0,0 +1,131 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2019 Looker Data Sciences, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+import { Looker40SDK as LookerSDK, IDashboard, IRenderTask } from '@looker/sdk'
+
+/**
+ * Find a dashboard by title
+ * @param sdk Looker SDK object
+ * @param {string} title of dashboard
+ * @returns {Promise<IDashboard>} the matched dashboard
+ */
+export const getDashboard = async (sdk: LookerSDK, title: string) => {
+  const [dash] = await sdk.ok(sdk.search_dashboards({title}))
+  if (!dash) {
+    console.warn(`No dashboard titled "${title}" was found`)
+  }
+  return dash
+}
+
+/**
+ * Get a tile by title from a dashboard
+ * @param {IDashboard} dash Dashboard to search
+ * @param {string} title Title title to find
+ * @returns {IDashboardElement | undefined} Returns the found tile or undefined
+ */
+export const getDashboardTile = (dash: IDashboard, title: string) => {
```

Comment:
Interesting to do this look up by title. I don't think those are guaranteed unique.

Reply:
Nope, they're not. I make no guarantees with the sample code. Just intended to demonstrate SDK/API usage and downloading.

Reply:
You did inspire me to list the available dashboards by ID and title when the requested dashboard isn't found, though